core/src/main/scala/kanaloa/dispatcher/Dispatcher.scala

```diff
@@ -22,7 +22,9 @@ trait Dispatcher extends Actor with ActorLogging {
   def settings: Dispatcher.Settings
   def backend: Backend
   def resultChecker: ResultChecker
-  def metricsCollector: ActorRef
+  def reporter: Option[Reporter]
+
+  protected lazy val metricsCollector = context.actorOf(MetricsCollector.props(reporter, settings.performanceSamplerSettings))
```

Comment:
I agree. Dependency injection while maintaining actor supervision hierarchy is annoying. In this particular case, `Dispatcher` does not have direct interaction with the `MetricsCollector` it merely inject it as dependency into the sub components, with which the interaction between those and `MetricsCollector` are tested. Thus I didn't find the need of having DI factory functions in this case.